node-db/week-13/api-workshop.md

```diff
@@ -0,0 +1,540 @@
+![](https://img.shields.io/badge/status-review-orange.svg)
+
+# Where we used APIs?
+
+When we called the Github _API_ to get the number of Repos associated to a user:
+
+```js
+var url = "https://api.github.com/users/CodeYourFuture/repos";
+
+fetch(url)
+  .then(function(data) {
+    return data.json();
+  })
+  .then(function(repos) {
+    var reposList = document.querySelector("#repos-list");
+    reposList.innerText = repos.length;
+  });
+```
+
+What happens in all these scenarios? What is the URL `https://api.github.com/users/CodeYourFuture/repos`? And where does it "live"? Is it on your computer the _client_, or on a remote _server_?
+
+Until now, we have been writing _client_ code (also called _client-side_ or _frontend_). The client in our case is the web browser (e.g. Firefox, Chrome etc.) and it talks to a _server_ (also called _server-side_ or _backend_). In the examples we have completed someone else wrote code to handle server _requests_ and provide a _response_.
+
+We are now taking the next step in our journey: to write _server-side_ code. JavaScript is the only language that the browser understands so you do not have options if you want to write code for the browser (for things like interacting with the DOM and what we have been doing so far).
+
+For the server there are a lot of programming languages you can use: PHP, Python, Java, Ruby, and many more. You'll be pleased to hear that you can also write JavaScript on the server so you don't have to learn another programming language if you don't want to! JavaScript on ther server is relatively new and only started with the introduction of _Node.js_ around 2009. Node.js is what we will use to write server-side code.
+
```

Comment:
Nothing to add the syllabus .. but during the class, at this point, we could discuss why not all code is in the browser? what kind of things are possible in the server but not browser and vice versa (I remember that came up and there was a lot of confusion about it)

Reply:
sure thing .. good point